ambersim/logger/logger.py

```diff
@@ -0,0 +1,146 @@
+import jax
+import wandb
+from torch.utils.tensorboard import SummaryWriter
+
+
+class BaseLogger:
+    """Base logger interface that defines common methods for logging metrics and parameters.
+
+    Attributes:
+        log_dir (str): Directory where logs are stored. If None, default locations are used.
+    """
+
+    def __init__(self, log_dir=None):
+        """Initializes the BaseLogger with a specified log directory.
+
+        Args:
+            log_dir (str): Directory to store the logs. If None, uses default log directory.
+        """
+        self.log_dir = log_dir
+
+    def log_metric(self, key, value, step=None):
+        """Logs a metric value.
+
+        Args:
+            key (str): The name of the metric.
+            value (float): The value of the metric.
+            step (int, optional): The step number at which the metric is logged.
+        """
+        raise NotImplementedError
```

Comment:
Question: do you want every child class to have `log_metric` and `log_params`? If so, I would make `BaseLogger` abstract and decorate these `abstractmethod` instead. If, however, there are some situations where child classes shouldn't necessarily implement these functions, then it's fine to leave it as-is. I don't know what vision we have for the logging API.